library/src-bootstrapped/scala/reflect/Generic.scala

```diff
@@ -0,0 +1,18 @@
+package scala.reflect
+
+/** A class for mapping between an ADT value and
+  *  the case mirror that represents the value.
+  */
+abstract class Generic[T] {
+
+  type Shape <: scala.compiletime.Shape
```

Comment:
Yes, or ideally a representation that can be used for both. Right now, I would like to learn first about possible use cases of compile-time labels. I could not come up with one that makes sense, without also postulating whitebox macros. For instance schema validation needs a whitebox macro to pull in a schema in the first place. So I opted for the simplest implementation that addresses the use cases I have right now. But if we get a good use case this is up for change.

Reply:
> For instance schema validation needs a whitebox macro to pull in a schema in the first place.

In Spark tables are typed as `Dataset[T]` where `T` is a case class. The [current API](https://spark.apache.org/docs/2.3.0/api/java/org/apache/spark/sql/Dataset.html#col-java.lang.String-) to reference table's columns is String based. With a type-level representation of case class labels this API could be made type-safe!

Also I think it might be enough to expose labels as `type`-s given that `valueOf` from #5647 allows users to summon values for singleton strings.

Reply:
Yes, Spark is one use case (there's even a project you might want to take a look at - https://github.com/typelevel/frameless).

Also generating lenses automatically is another use case I could come up with off the top of my head (prisms as well).